src/main/java/mate/academy/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public List<Order> getAllByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("select distinct o from Order o "
+                            + "inner join fetch o.tickets t "
+                            + "left join fetch t.user u "
+                            + "left join fetch t.movieSession ms "
+                            + "left join fetch ms.movie m "
+                            + "left join fetch ms.cinemaHall ch "
```

Comment:
is it possible to have ticket without user or movie session?

Reply:
Yea , it is , of course.